libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/ValidationUtils.kt

```diff
@@ -0,0 +1,44 @@
+package com.mapbox.services.android.navigation.v5.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import java.util.MissingFormatArgumentException
+
+class ValidationUtils {
+    companion object {
+        @JvmStatic
+        fun validDirectionsRoute(
+            directionsRoute: DirectionsRoute,
+            defaultMilestonesEnabled: Boolean
+        ) {
+            if (defaultMilestonesEnabled) {
+                val routeOptions = directionsRoute.routeOptions()
+                checkNullRouteOptions(routeOptions)
+                checkInvalidVoiceInstructions(routeOptions!!)
+                checkInvalidBannerInstructions(routeOptions)
+            }
+        }
+
+        private fun checkNullRouteOptions(routeOptions: RouteOptions?) {
+            if (routeOptions == null) {
+                throw MissingFormatArgumentException("Using the default milestones requires the " + "directions route to include the route options object.")
+            }
+        }
+
+        private fun checkInvalidVoiceInstructions(routeOptions: RouteOptions) {
+            val instructions = routeOptions.voiceInstructions()
+            val invalidVoiceInstructions = instructions == null || !instructions
```

Comment:
Should we rename and make it like
```suggestion
            val isValidVoiceInstructions = instructions != null && instructions
```
?
Cause it is bad practice to use negative names and then add `not` prefix in `if` conditions like `!invlid`... Much better is this condition will looks like `isValid`